### ISSUE TITLE
downloadUrl for a non-existent remote URL does not return a LookUpError

### ISSUE THREAD

**Issue by phil1995** [dropbox/dropbox-sdk-obj-c#263]
When I call the function `downloadUrl(...)` with a URL for a non-existent file, I only get an `DBRequestHttpError` with the status code 409, although there should be a `DBLookUpError` with a `pathNotFound`. (As this is also the case with `getMetadata(...)` for example).
If needed here are some requestIDs for `downloadURL(...)`, which all failed with status code 409:
```
DropboxHttpError[{
    ErrorContent = nil;
    RequestId = 2af78e7b5c29ca0d6947b3dd3be2d370;
    StatusCode = 409;
    UserMessage = nil;
}];

DropboxHttpError[{
    ErrorContent = nil;
    RequestId = b1cb58381ab889b7f31889ee8ac21761;
    StatusCode = 409;
    UserMessage = nil;
}];

DropboxHttpError[{
    ErrorContent = nil;
    RequestId = 35773acaed1b2c5c613087e44b39c5bb;
    StatusCode = 409;
    UserMessage = nil;
}];
```

**Comment by greg-db** [comment]
Thanks for the report! We'll look into it.

Also, for reference, do you happen to also get any `NSURLConnection` error output when this occurs for you?

**Comment by phil1995** [comment]
Sorry for the late reply.
Yes, I receive the following output:
`nil host used in call to allowsSpecificHTTPSCertificateForHost`
and
`NSURLConnection finished with error - code -1002`

**Comment by greg-db** [comment]
Thanks for confirming that!

This issue hasn't been addressed yet, but it's still open with engineering. I'll follow up here once I have an update on it.

**Comment by YufeiG** [comment]
> nil host used in call to allowsSpecificHTTPSCertificateForHost
 People seem to be hitting it last year on the iOS 14 beta (not dropbox related) but it should be resolved.

> NSURLConnection finished with error - code -1002
Looks like invalid URL error. NSURLErrorUnsupportedURL

>  DBRequestHttpError with the status code 409
409 | Conflict | The request cannot be accepted because the JSON data you have provided conflicts with rules enforced by the API.

I am wondering if the url you are sending is invalid in some way, not that is it non-existent. If you still remember, can you share an example path that you used and the code around how you made the request?

**Comment by greg-db** [comment]
This should be fixed in v6.0.1.
